pkg/api/scylla/validation/scylladbmanagerclusterregistration_validation.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2025 ScyllaDB
+
+package validation
+
+import (
+	"fmt"
+
+	scyllav1alpha1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1alpha1"
+	"github.com/scylladb/scylla-operator/pkg/naming"
+	apimachineryvalidation "k8s.io/apimachinery/pkg/api/validation"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+var (
+	supportedLocalScyllaDBReferenceKinds = []string{
+		naming.ScyllaDBDatacenterKind,
+	}
+)
+
+func ValidateScyllaDBManagerClusterRegistration(smcr *scyllav1alpha1.ScyllaDBManagerClusterRegistration) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, ValidateScyllaDBManagerClusterRegistrationObjectMeta(&smcr.ObjectMeta, field.NewPath("metadata"))...)
+	allErrs = append(allErrs, ValidateScyllaDBManagerClusterRegistrationSpec(&smcr.Spec, field.NewPath("spec"))...)
+
+	return allErrs
+}
+
+func ValidateScyllaDBManagerClusterRegistrationObjectMeta(objectMeta *metav1.ObjectMeta, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	globalScyllaDBManagerLabelValue, hasGlobalScyllaDBManagerLabel := objectMeta.Labels[naming.GlobalScyllaDBManagerLabel]
+	if !hasGlobalScyllaDBManagerLabel {
+		allErrs = append(allErrs, field.Required(fldPath.Child("labels").Key(naming.GlobalScyllaDBManagerLabel), ""))
+	} else if globalScyllaDBManagerLabelValue != naming.LabelValueTrue {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("labels").Key(naming.GlobalScyllaDBManagerLabel), globalScyllaDBManagerLabelValue, fmt.Sprintf("must be %q", naming.LabelValueTrue)))
+	}
+
+	return allErrs
+}
+
+func ValidateScyllaDBManagerClusterRegistrationSpec(spec *scyllav1alpha1.ScyllaDBManagerClusterRegistrationSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
```

Comment:
nit

note https://github.com/scylladb/scylla-operator/pull/2598 - this review inspired me to make that suggestion, please decide & get this PR in sync with that proposal.

Reply:
done